### PR TITLE
Add a Contributing Guidelines file

### PR DESCRIPTION
**Pull request by gdams**
We should have a contributing guidelines file:

Inital Draft, open to ideas

### CODE REVIEW

**Comment by gdams** [comment]
https://github.com/nodejs/citgm/issues/313

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+## Flaky and Skip Changes to lookup.json
+The process for adding flaky or skip tags to the lookup.json is reasonably simple. Make your changes put together a PR with ideally a link to the community CI failing and then it will be reviewed.
+
+
+## Submitting a module to CitGM
```
Comment (gibfahn) [inline CONTRIBUTING.md]: I don't think we'd want to duplicate this, a link to the `README` would work (and I think the readme is the best place for this stuff, it's not just for contributors).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+OR
+* The module is identified as an important module by a Node.js Working Group
+
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
+
+* Create an issue prior to submitting PR's as this can help speed up the process
+* Include tests for your code wherever possible
+* Ensure that `npm test` passes before submitting the PR
+* Please squash commits all commits into one single commit
```
Comment (gibfahn) [inline CONTRIBUTING.md]: ~commits~ all commits

Also I don't think it's necessary to squash everything into a single commit. It should be squashed into a single commit per logical change as part of landing, why does it need to be submitted that way? I know some people like to amend their existing commits, and some people like to push `fixup!` commits to preserve history in the PR.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* The module must cover unique portions of our API
+OR
+* The module fits into a key category (e.g. Testing, Streams, Monitoring, etc.)
+OR
+* The module is under the Node.js foundation Github org
+OR
+* The module is identified as an important module by a Node.js Working Group
+
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
```
Comment (gibfahn) [inline CONTRIBUTING.md]: I think general info should come first, and specific stuff like `Flaky and Skip Changes to lookup.json` should come afterwards.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+* Include any documentation or manual changes where necessary
+* Ensure that `npm test` passes before submitting the PR
+* Please squash each logical change into one single commit
+* Follow the nodejs commit guidelines shown [here](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit)
+
+## Flaky and Skip Changes to lookup.json
+The process for adding flaky or skip tags to the lookup.json is reasonably simple. Make your changes put together a PR with ideally a link to the community CI failing and then it will be reviewed.
+
+
+## Submitting a module to CitG
```
Comment (gdams) [inline CONTRIBUTING.md]: CitG ==> CitGM

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+* Create an issue prior to submitting PR's as this can help speed up the process
+* Include tests for your code wherever possible
+* Include any documentation or manual changes where necessary
+* Ensure that `npm test` passes before submitting the PR
+* Please squash each logical change into one single commit
+* Follow the nodejs commit guidelines shown [here](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit)
+
+## Flaky and Skip Changes to lookup.json
+The process for adding flaky or skip tags to the lookup.json is reasonably simple. Make your changes put together a PR with ideally a link to the community CI failing and then it will be reviewed.
```
Comment (gibfahn) [inline CONTRIBUTING.md]: This paragraph is mostly fluff. Also everything needs to be wrapped to <80chars (some editors do it automatically).

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
+
+* Create an issue prior to submitting PR's as this can help speed up the process
+* Include tests for your code wherever possible
+* Include any documentation or manual changes where necessary
+* Ensure that `npm test` passes before submitting the PR
+* Please squash each logical change into one single commit
```
Comment (gibfahn) [inline CONTRIBUTING.md]: `one single` -> `a single`

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
+
+* Create an issue prior to submitting PR's as this can help speed up the process
```
Comment (gibfahn) [inline CONTRIBUTING.md]: `PR's`->`PRs`

Comment (MylesBorins) [inline CONTRIBUTING.md]: or `a PR`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+* Include tests for your code wherever possible
+* Include any documentation or manual changes where necessary
+* Ensure that `npm test` passes before submitting the PR
+* Please squash each logical change into one single commit
+* Follow the nodejs commit guidelines shown [here](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit)
```
Comment (MylesBorins) [inline CONTRIBUTING.md]: we have not been following these commit guidelines. Do we want to start doing so? This feels like it should likely be it's own issue to discuss

File: CONTRIBUTING.md (hunk 5 above)
Comment (MylesBorins) [inline CONTRIBUTING.md]: I agree, this should be removed

**Comment by gdams** [comment]
@MylesBorins @gibfahn Updated PTAL

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
+
+* Create an issue prior to submitting a PR as this can help speed up the process
+* Include tests for your code wherever possible
+* Include any documentation or manual changes where necessary
```
Comment (gibfahn) [inline CONTRIBUTING.md]: `manual`->`man page` I think, manual here sounds like it means by hand (e.g. manual labour)

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+* Ensure that `npm test` passes before submitting the PR
+* Please squash each logical change into a single commit
+* Follow the nodejs commit guidelines shown [here](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit)
```
Comment (gibfahn) [inline CONTRIBUTING.md]: As @MylesBorins said in https://github.com/nodejs/citgm/pull/314#discussion_r96458927, we should probably discuss whether we're going to follow these guidelines. Looking at [recent commits](https://github.com/nodejs/citgm/commits/master) I guess we have been getting closer to the nodejs core style in terms of submitting PRs, we just haven't been adding any of the metadata on landing. 

So technically asking people to follow the commit guidelines is fine, we're using 50/72 chars and a `subsystem:` (we're just not adding a `PR-URL:` or `Reviewed-by:` on landing).

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
+
+* Create an issue prior to submitting a PR as this can help speed up the process
```
Comment (gibfahn) [inline CONTRIBUTING.md]: `Create`->`Consider creating`, I don't think we want an issue per PR as a hard requirement (e.g. for marking modules as flaky).

File: CONTRIBUTING.md (hunk 10 above)
Comment (gdams) [inline CONTRIBUTING.md]: So yeah I agree with @gibfahn, I personally feel that on a smaller scale project like this, we don't perhaps need to include `Reviewed-by:` but I can see that including `PR-URL:` may be useful.

Comment (gibfahn) [inline CONTRIBUTING.md]: Maybe worth just including the basic rules in a few bullet points

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
+## Making changes to CitGM
+#### There are a few basic requirements for making changes to CitGM
+
+* Consider creating an issue prior to submitting a PR as this can help speed up the process
+* Include tests for your code wherever possible
+* Include any documentation or man page changes where necessary
+* Ensure that `npm test` passes before submitting the PR
+* Please squash each logical change into a single commit
+* Follow the nodejs commit guidelines shown [here](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit)
```
Comment (gibfahn) [inline CONTRIBUTING.md]: https://github.com/nodejs/citgm/pull/314#discussion_r98095427 is still outstanding

>Maybe worth just including the basic rules in a few bullet points

**Comment by gibfahn** [comment]
Also can you change the commit subsystem from `docs:` to `doc:`?

**Comment by gibfahn** [comment]
Incidentally, given that [node-review](https://github.com/evanlucas/node-review) works in citgm, it's pretty straightforward to include the `PR-URL:` and `Reviewed-By:` labels in PRs. I think including `PR-URL:` is definitely worth it for landing other people's commits.

**Comment by MylesBorins** [comment]
@gibfahn we originally decided not to include all the meta data to make it a lower barrier to entry for contributing. I'm open to introducing it if @nodejs/ctigm is on board

**Comment by MylesBorins** [comment]
@gibfahn one thing to mention, node-review will only count LGTM's from people who are a core collaborator, since it is hardcoded. The fact that it "mostly" works is a happy accident

**Comment by gibfahn** [comment]
@MylesBorins good points. Okay, so how about optionally using `PR-URL:` and not bothering with the rest?

**Comment by MylesBorins** [comment]
Sgtm

**Comment by gdams** [comment]
SGTM

**Comment by gdams** [comment]
@nodejs/citgm I think this is ready, could I get some reviews please?

**Comment by MylesBorins** [comment]
I feel like the commit guidelines are overkill.if we are not generating a changelog do we really need anything more than a descriptive title?

**Comment by gibfahn** [comment]
>I feel like the commit guidelines are overkill.

Which ones exactly? The 50 (imperative verb), leave a line, 72 is just the standard Git commit guidelines AFAIK. Subsystem we've been doing already.

Other than that are there any other rules?

**Comment by targos** [comment]
I think it's good to have some guidelines, even if we do not generate a change log.

**Comment by MylesBorins** [comment]
I'll concede. I was simply thinking that anything more than asking for a
descriptive title was a bit much.

**Comment by sxa555** [comment]
At the moment when a PR is created it points you to the guidelines at https://github.com/nodejs/citgm/blob/master/CONTRIBUTING.md - can I suggest removing that until this has landed?

**Comment by gdams** [comment]
@sxa555 this is about to be landed, no need to worry

**Comment by gdams** [comment]
@nodejs/citgm are we all happy with this now ?

**Comment by gdams** [comment]
@MylesBorins that leaves you then?

**Pull request merged**